src/Mvc/Mvc.Core/test/Infrastructure/ActionResultTypeMapperTest.cs

```diff
@@ -23,6 +24,22 @@ public void Convert_WithIConvertToActionResult_DelegatesToInterface()
         Assert.Same(expected, result);
     }
 
+    [Fact]
+    public void Convert_WithIResult_DelegatesToInterface()
```

Comment:
Let's verify that we use the `IConvertToActionResult` implementation for types that implement both `IConvertToActionResult` and `IResult`